app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMapBackground.java

```diff
@@ -0,0 +1,214 @@
+package com.nexenio.bleindoorpositioningdemo.ui.beaconview.map;
+
+import android.graphics.Bitmap;
+import android.graphics.Point;
+import android.support.annotation.NonNull;
+
+import com.nexenio.bleindoorpositioning.location.Location;
+import com.nexenio.bleindoorpositioning.location.projection.CanvasProjection;
+
+public class BeaconMapBackground {
+
+    private Bitmap imageBitmap;
+
+    private double metersPerPixel;
+
+    private double bearing;
+
+    private Location topLeftLocation;
+    private Location bottomRightLocation;
+
+    private Point topLeftPoint;
+    private Point bottomRightPoint;
+
+    private CanvasProjection projection = new CanvasProjection();
+
+    private BeaconMapBackground(Bitmap imageBitmap) {
+        this.imageBitmap = imageBitmap;
+        projection.setCanvasWidth(imageBitmap.getWidth());
+        projection.setCanvasHeight(imageBitmap.getHeight());
+        topLeftPoint = new Point(0, 0);
+        bottomRightPoint = new Point(imageBitmap.getWidth(), imageBitmap.getHeight());
+    }
+
+    public Location getLocation(Point point) {
+        return getLocation(point, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public Point getPoint(Location location) {
+        return getPoint(location, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public static float getMetersPerPixel(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double distanceInPixels = getPixelDistance(firstReferencePoint, secondReferencePoint);
+        double distanceInMeters = firstReferenceLocation.getDistanceTo(secondReferenceLocation);
+        return (float) (distanceInMeters / distanceInPixels);
+    }
+
+    public static double getBearing(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double locationBearing = getBearing(firstReferenceLocation, secondReferenceLocation);
+        double pointBearing = getBearing(firstReferencePoint, secondReferencePoint);
+        return ((locationBearing - pointBearing) + 360) % 360;
+    }
+
+    public static double getBearing(Point firstPoint, Point secondPoint) {
+        double angle = Math.atan2(secondPoint.y - firstPoint.y, secondPoint.x - firstPoint.x) * 180 / Math.PI;
```

Comment:
`Atan2` has a value range from `Pi` to `-Pi`. If we divide by `Pi` we will have a range of -1 to 1. Multiplying with 180 gives -180 to 180. 
If we now add 90 we will be in a range from -90 to 270. Modulo 360 will leave the value range in between -90 and 270.

Reply:
There's a test that asserts a value > 270:

```java
@Test
public void getBearing() {
    double bearing = BeaconMapBackground.getBearing(firstReferenceLocation, firstReferencePoint, secondReferenceLocation, secondReferencePoint);
    assertEquals(353.84, bearing, 0.001);
    assertEquals(bearing, beaconMapBackground.getBearing(), 0.001);
}
```

Other implementations ([1](https://gist.github.com/conorbuck/2606166) and [2](https://stackoverflow.com/questions/46590154/calculate-bearing-between-2-points-with-javascript)) look equal to the current one.